lib/rucio/core/rule.py

```diff
@@ -2868,10 +2868,11 @@ def __delete_lock_and_update_replica(lock, purge_replicas=False, nowait=False, s
                 replica.tombstone = replica.accessed_at
             else:
                 replica.tombstone = replica.created_at
-        if lock.state == LockState.REPLICATING and replica.lock_cnt == 0:
-            return True
+            if lock.state == LockState.REPLICATING:
+                replica.state = ReplicaState.UNAVAILABLE
```

Comment:
Why do you set the replica state to 'U' whereas it was not the case before?

Reply:
It's just a re-arrange of the conditions as it looks simpler like this. It was set to UNAVAILABLE before too, it just happened in the lines 2860+2861 before.